🔥Linting on Fire 🔥

AlmogCohen opened the pull request:
fix many style issues
fix flake8: E127, E128, E302, E303, E731, W391
fix pylint errors: trailing-whitespace, trailing-newlines

MANY files have changed. Hope it is helpful, please inspect carefully although I've tried to make no logical changes to the code :) Enjoy!

AlmogCohen commented:
And please accept my other open PR before this one so I don't have to handle all the conflicts!!! :)

mscuthbert commented:
Can you update from master first so that you get the GitHub actions checks running on this? Thanks!

mscuthbert commented:
You might need to close this PR and make a new one. I’m not 100% clear on how the new GitHub actions work when the script has updated.

mscuthbert commented:
I don’t find trailing new lines a problem. What problem do they cause that would ever be worth making someone wait 15 minutes for a new check?

AlmogCohen commented:
1. I just checked and I'm pretty sure that I'm 100% on top of master, fully merged from master. Not sure why the checks are not running. Maybe they run only for owners? Not sure if reopening would do anything but I can try if you think so.
2. Trailing new lines: My take on it: Good developers will **not** commit a PR just to see what travis has to say about it. Most style and unit tests should be run locally by contributors. Travis is only a last-resort safety check for the repo owners to see that everything was done right. On the other hand, good repos will have clear instructions in their README for how to run linting and unit testing locally (which we are still lacking here). Generally I go with PEP8, unless it is a real issue for the project (like the camelCase in music21 which no sane person would enforce blindly).

AlmogCohen commented:
The commit of this PR is just on top of the latest master commit https://share.getcloudapp.com/jkuy6dRn

Anything I can help with? Maybe it something with the GitHub settings for GitHub checks?

AlmogCohen commented:
@mscuthbert I've just also simplified the command for running `flake8`. From now on you simply run `flake8` and all checks are runnings (as opposed to `flake8 music21` we had before. Following that I've fixed some issues in `setup.py` and the `installer.py`.

If you are open to that, I can add documentation for linting into the main README. You can see an example for a Developing/Contribution section in a README file in another project I authored: https://github.com/algosec/algosec-python#developing

If you are open to that, we can document the `flake8` as well as `pylint` to improve collaboration in this project. One step after would be to document how to run all the unittests locally. All commands should hopefully be one word. What do you think? I'd be happy to continue helping if you feel like it

AlmogCohen commented:
But yes, trailing new lines doesn't seem so important :) 

I looked up [this](https://softwareengineering.stackexchange.com/a/121560) but am not convinced. We can disable that again if you wish

mscuthbert commented:
Hi!  I'm at the Society for Music Theory conference so won't be able to reply until Wednesday or so.  But yeah, I've become convinced on trailing whitespace as a problem because when someone else edits the code it's much harder to see what lines actually changed.  But that's not generally a problem with trailing new lines.  

Basically my view (unlike, say "Black") is I want maximum freedom that doesn't cause problems for me and others.

And over 12 years the vast majority of submissions to music21 have been by me, so the code style has to be something that **I** enjoy using and seeing.  :-)

AlmogCohen commented:
You are right and I'm happy to comply, thanks for the feedback, enjoy the
conference :) Let me know which errors you want me to ignore again besides
the trailing new lines as well as any other changes to the repo.

Thanks for the great work!

On Fri, Nov 8, 2019, 21:23 Michael Scott Cuthbert <notifications@github.com>
wrote:

> Hi! I'm at the Society for Music Theory conference so won't be able to
> reply until Wednesday or so. But yeah, I've become convinced on trailing
> whitespace as a problem because when someone else edits the code it's much
> harder to see what lines actually changed. But that's not generally a
> problem with trailing new lines.
>
> Basically my view (unlike, say "Black") is I want maximum freedom that
> doesn't cause problems for me and others.
>
> And over 12 years the vast majority of submissions to music21 have been by
> me, so the code style has to be something that *I* enjoy using and
> seeing. :-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cuthbertLab/music21/pull/474?email_source=notifications&email_token=AA5VOZSC7JBTBSLE6XJJJEDQSW4DPA5CNFSM4JKEWIE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDTDK6Y#issuecomment-551957883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5VOZVU6QMCUZO542I7VQTQSW4DPANCNFSM4JKEWIEQ>
> .
>

mscuthbert commented on music21/abcFormat/testFiles.py:
```diff
@@ -641,26 +637,20 @@
-    '''
-    Get test material by type of content
-    '''
+    """Get test material by type of content"""
```
this is incorrect -- sorry.  Music21 house style is single quoted docstrings with a new line before and after.

mscuthbert commented on music21/alpha/analysis/hasher.py:
```diff
@@ -431,17 +431,23 @@ def hashStream(self, s):
                     for n in elt:
-                        singleNoteHash = [self.hashingFunctions[hashProperty](n, thisChord=elt)
-                                            for hashProperty in self.tupleList]
+                        singleNoteHash = [
```
I prefer the other style.

mscuthbert commented on music21/analysis/discrete.py:
```diff
@@ -568,33 +560,33 @@ def _bestKeyEnharmonic(self, pitchObj, mode, sStream=None):
-#             favor = 'flat'
-#         else:
-#             favor = None
+        #         if self.sharpFlatCount is None:
```
when indenting comments, the spaces on the other side of the `#` must be removed.

mscuthbert commented:
yeah, I should Squash when there are tons of commits.  But I don't believe it actually makes the `.git` file any smaller, since the individual commits are still there -- they can still be cherry picked out I believe.

AlmogCohen commented:
Thanks for the feedback. I totally understand. I'll be happy to gradually contribute following discussions of the matter
